### PR TITLE
Remove space around power operator for better readability

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Following PEP-8

> If operators with different priorities are used, consider adding whitespace around the operators with the lowest priority(ies). Use your own judgment;

https://www.python.org/dev/peps/pep-0008/#other-recommendations

I've applied this selectively where it makes sense. For example, I did not remove space for
 `base ** np.ceil(np.log(x) / np.log(base))`.

### CODE REVIEW

File: examples/images_contours_and_fields/tricontour_demo.py [hunk 1]
```diff
@@ -75,7 +75,7 @@
 y = np.degrees(xy[:, 1])
 x0 = -5
 y0 = 52
-z = np.exp(-0.01 * ((x - x0) ** 2 + (y - y0) ** 2))
+z = np.exp(-0.01 * ((x - x0)**2 + (y - y0)**2))
```
Comment (anntzer) [inline examples/images_contours_and_fields/tricontour_demo.py]: generally (throughout the pr) it feels wrong to me to have no spaces around the power when there are spaces within the operand (`(x - x0)`).

Comment (timhoffm) [reply examples/images_contours_and_fields/tricontour_demo.py]: Given the possible alternatives
~~~
((x - x0) ** 2 + (y - y0) ** 2))

((x - x0)**2 + (y - y0)**2))

((x-x0)**2 + (y-y0)**2))
~~~
I find the given one the best; the power is an attachment to the parenthesis. Spaces everywhere do not give any guidance. No spaces in the parentheses is sort of ok if the operand is as simple as here.

Comment (anntzer) [reply examples/images_contours_and_fields/tricontour_demo.py]: It feels rather silly to discuss this here, but https://www.python.org/dev/peps/pep-0008/#other-recommendations does say yes for `(a+b) * (a-b)` and no for `(a + b) * (a - b)`, which seems the closest to the current case (and which I agree with).

Comment (timhoffm) [reply examples/images_contours_and_fields/tricontour_demo.py]: I've removed spaces in the above case and similar cases. Rule of thumb: No spaces between most inner operators the there are multiple of these terms.

I know now why I do not want spaces around the power operator: In typeset math, you will have spacing around all regular operators, but the power superscript is directly attached to it's operand, whether it's a single variable or a bracketed term.

Comment (anntzer) [reply examples/images_contours_and_fields/tricontour_demo.py]: To take an example further down, `(0.2 + (ym - yy) / ym)**2` really looks wrong to me (even with the parenthesis, the power really seems to bind to `ym`).

Comment (WeatherGod) [reply examples/images_contours_and_fields/tricontour_demo.py]: My 2 cents: I agree with @timhoffm to remove the spaces around the power operator, especially when the term is being raised to an integer power.

Comment (timhoffm) [reply examples/images_contours_and_fields/tricontour_demo.py]: > `(0.2 + (ym - yy) / ym)**2` really looks wrong to me

Long terms are hard to read anyway and I can't argue against personal impression. So what are the options:

- Removing all the inner spaces makes it unreadble
  `(0.2+(ym-yy)/ym)**2`
- Having spaces around the power operator with short arguments detaches the exponent
  `(0.2 + (ym - yy) / ym) ** 2`
- IMHO the proposed version is still the best; and it's similar to the TeX rendered version:
   `(0.2 + (ym - yy) / ym)**2`
  grafik
  give or take spaces around the division operator.

I'm not claiming to be able to solve all readability issues.

Comment (anntzer) [reply examples/images_contours_and_fields/tricontour_demo.py]: I don't feel strongly enough to block this, but let's see what others think.

**Comment by timhoffm** [comment]
This has stalled and redoing this from scratch is probably easier than rebasing.

I suppose, we could at least agree on simple cases like `x**2` being preferable to  `x ** 2`. But I'm not going to do the work again right now.

**Pull request closed without merging**